Lib/importlib/abc.py

```diff
@@ -382,23 +388,35 @@ def read_text(self, encoding=None):
     def is_dir(self) -> bool:
         """
         Return True if self is a dir
+
+        If the directory/file cannot be found, FileNotFoundError should be raised,
```

Comment:
I'm not sure `FileNotFoundError` is correct here. Pathlib returns False when self does not exist.

```
>>> pathlib.Path('./does-not-exist').is_dir()
False
```